contracts/rust/adapter/src/sol_types.rs

```diff
@@ -201,3 +201,186 @@ impl From<G1PointSol> for staketable::BN254::G1Point {
         unsafe { std::mem::transmute(v) }
     }
 }
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use self::{
+    staketable::{EdOnBN254::EdOnBN254Point, BN254::G2Point},
+    StakeTable::{
+        ConsensusKeysUpdated, Delegated, Undelegated, ValidatorExit, ValidatorRegistered,
+    },
+};
+
+impl PartialEq for ValidatorRegistered {
```

Comment:
This is weird. This event has an empty `derive` in the bindings now.

Reply:
Yeah, think its unable to figure out if `G2Point` and `EdOnBN254Point` have PartialEq, maybe because the associated `RustType` does not have PartialEq? i.e SolTypeValue does not have PartialEq bound? idk